snap/naming/componentref.go

```diff
@@ -75,3 +76,16 @@ func (cid *ComponentRef) UnmarshalYAML(unmarshall func(interface{}) error) error
 
 	return nil
 }
+
+// snapPackComponentFilename is a regular expression that matches what snap pack
+// creates when building a component. For example, "foo+bar_1.0.0.comp" matches
+// this expression.
+var snapPackComponentFilename = regexp.MustCompile(fmt.Sprintf(`^(%[1]s)\+(%[1]s)(?:_.*)?\.comp$`, almostValidNameRegexString))
+
+func ComponentRefFromSnapPackFilename(filename string) (ComponentRef, error) {
+	matches := snapPackComponentFilename.FindStringSubmatch(filename)
+	if len(matches) != 3 {
+		return ComponentRef{}, fmt.Errorf("cannot parse snap pack component filename: %q", filename)
+	}
+	return NewComponentRef(matches[1], matches[2]), nil
```

Comment:
Should this validate the version? (`ValidateVersion`?)

Reply:
naming can't import snap, where ValidateVersion lives unfortunately. I'm not sure if it worth it here to fix that import cycle. Regardless, this isn't really trying to validate anything, more just a best-effort way of getting the right component name.